src/libcollectdclient/client.c

```diff
@@ -452,9 +463,11 @@ static int lcc_open_netsocket(lcc_connection_t *c, /* {{{ */
     }
   }
 
-  struct addrinfo ai_hints = {.ai_family = AF_UNSPEC,
-                              .ai_flags = AI_ADDRCONFIG,
-                              .ai_socktype = SOCK_STREAM};
+  struct addrinfo ai_hints = {
+      .ai_family = AF_UNSPEC, .ai_flags = 0, .ai_socktype = SOCK_STREAM};
+#ifdef AI_ADDRCONFIG
+  ai_hints.ai_flags = AI_ADDRCONFIG;
```

Comment:
could you define AI_ADDRCONFIG to 0 if not previously defined? that would clean this up nicely.

Reply:
Done.